jobserver/authorization/decorators.py

```diff
@@ -0,0 +1,26 @@
+from functools import wraps
+
+from django.http import Http404
+
+from .roles import CoreDeveloper
+from .utils import has_role
+
+
+def require_role(role):
+    """Decorator for views which require a given Role"""
+
+    def decorator_require_role(f):  # sigh
+        @wraps(f)
+        def wrapper_require_role(request, *args, **kwargs):
+            if has_role(request.user, role):
+                return f(request, *args, **kwargs)
+
+            # TODO: shoudl this redirect to / with a message?
+            raise Http404()
```

Comment:
Hmm, feel like this should probably be a 403, unless leaking the existance of something is a problem here?

Reply:
I thought it could have been but I can't come up with a good reason for it now which shows there likely wasn't one!

Fixed and rebased in.